docs: update stringify docs (#418)
asyncapi/parser-js: README.md

Converting a parsed document to a string may be necessary when saving the parsed document to a database, or similar situations where you need to parse the document just once and then reuse it.

> For that, the Parser supports the ability to stringify a parsed AsyncAPI document through the static `AsyncAPIDocument.stringify(...parsedDoc)` method. This method differs from the native `JSON.stringify(...json)` implementation, in that every reference that occurs (at least twice throughout the document) is converted into a [JSON Pointer](https://datatracker.ietf.org/doc/html/rfc6901) path with `$ref:` prefix, like:

Reviewer: ```suggestion
For that, the Parser supports the ability to stringify a parsed AsyncAPI document through the static `AsyncAPIDocument.stringify(...parsedDoc)` method. This method differs from the native `JSON.stringify(...json)` implementation, in that every reference that occurs (at least twice throughout the document) is converted into a [JSON Pointer](https://datatracker.ietf.org/doc/html/rfc6901) path with a `$ref:` prefix:
```

```json
{
  "foo": "$ref:$.some.path.to.the.bar"
}
```
		

> To parse a stringified document into an AsyncAPIDocument instance, you must use the static `AsyncAPIDocument.parse(...stringifiedDoc)` method. It isn't compatible with the native `JSON.parse()` method. It replaces the given references pointed by [JSON Pointer](https://datatracker.ietf.org/doc/html/rfc6901) path with `$ref:` prefix to the original objects.

Reviewer: ```suggestion
It replaces the given references pointed by the [JSON Pointer](https://datatracker.ietf.org/doc/html/rfc6901) path, with an `$ref:` prefix on the original objects. 
```

I **hope** this is correct? I am trying to make it sound better but unsure if I made it worse

Author: @alequetzalli  Hmm, I don't think so that `on` is better that `to`, because it should means that every occurrence of the JSON Pointer is changed to the object where JSON Pointer points, e.g.:

```json
{
  foo: '$ref:$.bar.foo',
  bar: {
    foo: {
      someKey: 'someValue'
    }
  }
}
``` 

so then `foo` key will be changed to the `bar.foo` value, so at the end we will have this:

```json
{
  foo: {
    someKey: 'someValue'
  },
  bar: {
    foo: {
      someKey: 'someValue'
    }
  }
}
```

I hope that it helps :)

Author: @alequetzalli :)

Reviewer: sorry, I did not see this notification yesterday 🤦‍♀️ 

Got it! I went and edited my previous commit suggestion, and left the original `to` like you had it before. 😀👍🏽👍🏽
